### PR TITLE
.NET auto-instrumentation include musl in docker image

### PR DESCRIPTION
**Pull request by Kielek**
Towards #1849

## Changes in the docker image
 - includes musl version
 - remove redundant, zip files from final image
 - execute commands in a batch, to reduce number of layers in final image

## Tests

Tested locally with defined spec->dotnet->image

```yaml
apiVersion: opentelemetry.io/v1alpha1
kind: Instrumentation
metadata:
  name: my-instrumentation
spec:
  exporter:
    endpoint: http://otel-collector:4317
  propagators:
    - tracecontext
    - baggage
    - b3
  sampler:
    type: parentbased_traceidratio
    argument: "0.25"
  python:
    env:
      # Required if endpoint is set to 4317.
      # Python autoinstrumentation uses http/proto by default
      # so data must be sent to 4318 instead of 4317.
      - name: OTEL_EXPORTER_OTLP_ENDPOINT
        value: http://otel-collector:4318
  dotnet:
    image: local-auto:dotnet
    env:
      # Required if endpoint is set to 4317.
      # Dotnet autoinstrumentation uses http/proto by default
      # See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/888e2cd216c77d12e56b54ee91dafbc4e7452a52/docs/config.md#otlp
      - name: OTEL_EXPORTER_OTLP_ENDPOINT
        value: http://otel-collector:4318
```

By application hosted on alpine linux. with manually redefined environmental variable (operator is not injecting variable if it is already in place).
```
        env:
          - name: "CORECLR_PROFILER_PATH"
            value: "/otel-auto-instrumentation/linux-musl-x64/OpenTelemetry.AutoInstrumentation.Native.so"
```

Checked also debian (glibc) based without any changes. It is working correctly.

## Notes

Changelog will be included into follow up PR with support for annotations.
~~I am marking this as [chore], as I do not have privileges to add `Skip Changelog` label. If possible, please change it.~~ Mikołaj, thanks.

It will be great to merge it even without follow up PR. It should unblock at least some scenarios and make the further development easier.

### CODE REVIEW

**Review by swiatekm-sumo — APPROVED** [review]
LGTM

**Review by lachmatt — APPROVED** [review]

**Review by iblancasa — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]
🥳

File: autoinstrumentation/dotnet/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,8 @@
 #  - Following environment variables are injected to the application container to enable the auto-instrumentation.
 #    CORECLR_ENABLE_PROFILING=1
 #    CORECLR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
-#    CORECLR_PROFILER_PATH=%InstallationLocation%/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
+#    CORECLR_PROFILER_PATH=%InstallationLocation%/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so # for glbic based images
```
Comment (pjanotti) [inline autoinstrumentation/dotnet/Dockerfile:8]:
```suggestion
#    CORECLR_PROFILER_PATH=%InstallationLocation%/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so # for glibc based images
```

**Review by jvoravong — APPROVED** [review]

**Comment by Kielek** [comment]
@open-telemetry/operator-maintainers, could you please merge it?
We have just release OTel Auto .NET 1.0.0. It will be great to include these changes before upgrading version.

**Review by yuriolisa — APPROVED** [review]

**Pull request merged**